trades/create_channel_sale.js

```diff
@@ -0,0 +1,237 @@
+const asyncAuto = require('async/auto');
+const asyncReflect = require('async/reflect');
+const {returnResult} = require('asyncjs-util');
+const {getNetwork} = require('ln-sync');
+const {getChainFeeRate} = require('ln-service');
+const {getChannels} = require('ln-service');
+const {getWalletInfo} = require('ln-service');
+const {signMessage} = require('ln-service');
+const {randomBytes} = require('crypto');
+const createAnchoredTrade = require('./create_anchored_trade');
+const serviceOpenTrade = require('./service_open_trade');
+
+// const serviceChannelSale = require('../sales/service_channel_sale');
+
+const asNumber = n => parseFloat(n, 10);
+const defaultExpirationDays = 1;
+const daysAsMs = days => Number(days) * 1000 * 60 * 60 * 24;
+const futureDate = ms => new Date(Date.now() + ms).toISOString();
+const isNumber = n => !isNaN(n);
+const slowTarget = 1000;
+const saleCost = (amount, rate) => (amount * rate / 1000000).toFixed(0);
+const saleSecret = randomBytes(48).toString('hex');
+const sellAction = 'sell';
+const tradeDescription = (alias, capacity) => `channelsale:${alias}-${capacity}`;
```

Comment:
should use tlv records to encode data, the anchor invoice already has these records and they can be extended